net/v2ray-core/Makefile

```diff
@@ -0,0 +1,127 @@
+#
+# Copyright (C) 2019 Xingwang Liao
+# Copyright (C) 2019 Alonzo Liu <unl89@protonmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=v2ray-core
+PKG_VERSION:=4.20.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/v2ray/v2ray-core/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=579a7098b8f6cc68c3f94c5453fa274299f8b072736f364654298bd4cf57d24e
+
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=Alonzo Liu <unl89@protonmail.com>
+
+PKG_CONFIG_DEPENDS := 
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
+
+GO_PKG:=v2ray.com/core
+
+include $(INCLUDE_DIR)/package.mk
+include $(TOPDIR)/feeds/packages/lang/golang/golang-package.mk
+
+define Package/v2ray-core/Default
+  TITLE:=A platform for building proxies
+  URL:=https://www.v2ray.com
+endef
+
+define Package/v2ray-core/Default/description
+Project V is a set of network tools that help you to build your own computer network.
+It secures your network connections and thus protects your privacy.
+endef
+
+define project-v/SubMenu
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Project V
+endef
+
+define Package/v2ray-core
+  $(call Package/v2ray-core/Default)
+  $(call project-v/SubMenu)
+  TITLE+= (full)
+  USERID:=v2ray=10800:v2ray=10800
+  DEPENDS:=$(GO_ARCH_DEPENDS) +ca-bundle
+endef
+
+define Package/v2ray-core/config
+endef
+
+define Package/v2ray-core/description
+$(call Package/v2ray-core/Default/description)
+
+  This package contains v2ray, v2ctl and v2ray-assets.
+endef
+
+define Build/Prepare
+	$(Build/Prepare/Default)
+	( \
+		sed -i \
+			's/\(version[[:space:]]*=[[:space:]]*"\).*\("\)/\1$(PKG_VERSION)\2/; \
+			s/\(build[[:space:]]*=[[:space:]]*"\).*\("\)/\1OpenWrt - Release $(PKG_RELEASE)\2/' \
+			$(PKG_BUILD_DIR)/core.go ; \
+	)
+	wget -O $(PKG_BUILD_DIR)/release/config/geoip.dat \
+		https://github.com/v2ray/geoip/releases/download/20190516.1/geoip.dat
+
+	wget -O $(PKG_BUILD_DIR)/release/config/geosite.dat \
+		https://github.com/v2ray/domain-list-community/releases/download/20190829.1/dlc.dat
```

Comment:
I still think you should use [download overrides](https://openwrt.org/docs/guide-developer/packages#download_override) for these:

* The files will be cached (along with other downloaded source tarballs)
* The downloads will be verified against their (sha256) checksums

You can see the [gcc package](https://github.com/openwrt/packages/blob/c722105c1d472bb89752fa732b23f27c8c5cb540/devel/gcc/Makefile#L84-L126) as an example. (For this package, copying the files from `$(DL_DIR)` into `$(PKG_BUILD_DIR)` during `Build/Prepare` should be enough.)

I suggest setting both `URL_FILE` (the original filename, from the URL) and `FILE` (filename to save locally) though, and include the version number/string in `FILE` (see [usbip](https://github.com/openwrt/packages/blob/c722105c1d472bb89752fa732b23f27c8c5cb540/net/usbip/Makefile#L85-L92) and [pciutils](https://github.com/openwrt/packages/blob/c722105c1d472bb89752fa732b23f27c8c5cb540/utils/pciutils/Makefile#L49-L56)).